### PR TITLE
create first schema transformation

### PR DESCRIPTION
**Pull request by mt-bt**

### CODE REVIEW

File: src/server/foundation/utils.clj [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+       (->> params
+            (map (fn [[key value]] (str (name key) " = "  (quote-unquote value))))
+            (interpose " and ")
+            (apply str))))
+
+(defn gen-lacinia-sch
+  [entity-name schema]
+  (let [resolver-id       (keyword (str "get-"(name entity-name)))
+        queryiable-fields (filter (comp :q second) schema)
+        query-function    (partial gen-query-handler entity-name)
```
Comment (PBRT) [inline src/server/foundation/utils.clj:40]: <3

File: src/server/foundation/utils.clj [hunk 2]
```diff
@@ -0,0 +1,49 @@
+(ns foundation.utils.lacinia
+  (:require [clojure.spec.alpha :as s]))
+
+; maybe use lacinia's own specs
+(comment (s/def ::type (s/or :scalar #{'Int 'String 'ID 'Float 'Boolean '}
```
Comment (PBRT) [inline src/server/foundation/utils.clj:5]: Let's see later how to share all the typing thing between lacinia/sql

File: src/server/tests/main.clj [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+             [foundation.utils.lacinia :as utils]))
+
+(def user
+  '{:id {:type Int
+         :q true}
+    :username {:type String
+               :q true}
+    :description {:type String}})
+
+(t/deftest "gen-query-handler: sql query generator."
```
Comment (PBRT) [inline src/server/tests/main.clj:12]: Neat! Those one will grow with the time

**Review by PBRT — APPROVED** [review]

File: src/server/foundation/utils.clj (hunk 2 above)
Comment (mt-bt) [reply src/server/foundation/utils.clj:5]: Yes I definitely see some benefits to use spec for those schemas transformations

**Pull request merged**